projects/core/src/cart/facade/cart.service.ts

```diff
@@ -223,4 +225,67 @@ export class CartService {
   private isLoggedIn(userId: string): boolean {
     return typeof userId !== 'undefined';
   }
+
+  // TODO: Remove once backend is updated
+  /**
+   * Temporary method to merge guest cart with user cart because of beackend limitation
+   * This is for an edge case
+   */
+  private guestCartMerge(): void {
+    let cartEntries: OrderEntry[];
+    this.getEntries()
+      .pipe(take(1))
+      .subscribe(entries => {
+        cartEntries = entries;
+      });
+
+    this.store.dispatch(
+      new CartActions.DeleteCart({
+        userId: ANONYMOUS_USERID,
+        cartId: this.cartData.cart.guid,
+      })
+    );
+
+    combineLatest([
+      this.store.select(CartSelectors.getCartContent),
+      this.store.select(CartSelectors.getCartLoading),
+    ])
+      .pipe(
+        filter(([, loading]) => !loading),
+        tap(([cart]) => {
+          // If the cart is not created it needs to be created
+          // This step should happen before adding entries to avoid conflicts in the requests
+          if (!this.isCreated(cart)) {
+            this.store.dispatch(
+              new CartActions.CreateCart({ userId: this.cartData.userId })
+            );
+          }
+        }),
+        filter(([cart]) => this.isCreated(cart)),
+        take(1)
+      )
+      .subscribe(() => {
+        this.store
```

Comment:
Can we move this part into another function "addEntries". I think maybe customers also need it; and it make this code more readable.